src/Tizen.MachineLearning.Inference/doc/api/Tizen.MachineLearning.Inference.md

```diff
@@ -0,0 +1,111 @@
+---
+uid: Tizen.MachineLearning.Inference
+summary: Machine learning features help you to handle machine learning frameworks, like TensorFlow-Lite and PyTorch, with the construction of a data pipeline. There are two main machine learning inference APIs. The first is the NNStreamer pipeline API, which provides interfaces to create and execute stream pipelines with neural networks and sensors. The second is the NNStreamer single API, which provides interfaces to invoke a neural network model with a single instance of input data.
+remarks: *content
+---
+## Overview
+There are two types of Machine Learning Inference API - Single API and Pipeline API.  
+Single API is useful for a simple usage scenario of neural network models. It allows the direct invoking of a neural network model with a single instance of input data for the model. It is useful if you have the input data pre-processed with the application itself and there are no complex interactions between neural network models, data processors, or data stream paths.  Pipeline API allows developers to construct and execute pipelines with multiple neural network models, multiple inputs and output nodes, multiple data processors, pre-and-post processors, and various data path manipulators. Besides, if the input is online data or streamed data, Pipeline API simplifies your application and improves its performance.
+ - Support various neural network frameworks (NNFW)  
+  TensorFlow, TensorFlow-Lite, Caffe2, PyTorch, and NNTrainer are the supported neural network frameworks. Neural network model files trained by such frameworks can be imported as filters of pipelines directly. Custom Filters, which are neural network models implemented directly with programming languages including C/C++ and Python, maybe imported as filters of pipelines directly as well.
+
+### Single API
+This section shows how to load a model without the construction of pipelines:  
+  1. Create single shot instance with model path:
+
+      ```cs
+      using Tizen.MachineLearning.Inference;
+
+      private TensorsInfo in_info;
+      private TensorsInfo out_info;
+      ...
+      var single = new SingleShot(_modelPath, in_info, out_info);
+      var single = new SingleShot(_modelPath, in_info, out_info, NNFWType.TensorflowLite, HWType.CPU, false);
+      var single = new SingleShot(_modelPath, NNFWType.Any, HWType.Any, true);
+      ```
+
+      The SingleShot instance can be created by using method overloading. If the given model has a flexible input/output dimension, `in_info` and `out_info` are required. `in_info` contains the information of the input tensors, and `out_info` contains the information of the output tensors.
```

Comment:
May change a word not to confuse with flexible tensors:
```suggestion
      The SingleShot instance can be created by using method overloading. If the given model has a dynamic input/output dimension, `in_info` and `out_info` are required. `in_info` contains the information of the input tensors, and `out_info` contains the information of the output tensors.
```

Reply:
Good, Updated.